File: packages/backend-adapters/django/rajutechie_streamkit_django/token.py
import time
import jwt
from django.conf import settings


def generate_user_token(
    user_id: str,
    name: str | None = None,
    role: str = "user",
    expires_in: int = 3600,
    grants: dict | None = None,
) -> str:
    api_key = getattr(settings, "RAJUTECHIE_STREAMKIT_API_KEY", "")
    api_secret = getattr(settings, "RAJUTECHIE_STREAMKIT_API_SECRET", "")

    now = int(time.time())
    payload = {
        "sub": user_id,
        "iss": api_key,
        "iat": now,
        "exp": now + expires_in,
        "role": role,
    }
    if name:
        payload["name"] = name
    if grants:
        payload["grants"] = grants

    return jwt.encode(payload, api_secret, algorithm="HS256")


def verify_token(token: str) -> dict:
    api_key = getattr(settings, "RAJUTECHIE_STREAMKIT_API_KEY", "")
    api_secret = getattr(settings, "RAJUTECHIE_STREAMKIT_API_SECRET", "")

    return jwt.decode(token, api_secret, algorithms=["HS256"], issuer=api_key)

File: packages/server-sdk/src/token.ts
import jwt from 'jsonwebtoken';

export interface TokenGeneratorConfig {
  apiKey: string;
  apiSecret: string;
}

export interface TokenClaims {
  userId: string;
  name?: string;
  role?: string;
  expiresIn?: string | number;
  grants?: Record<string, string[]>;
}

export class TokenGenerator {
  private apiKey: string;
  private apiSecret: string;

  constructor(config: TokenGeneratorConfig) {
    this.apiKey = config.apiKey;
    this.apiSecret = config.apiSecret;
  }

  generate(claims: TokenClaims): string {
    const payload: Record<string, unknown> = {
      sub: claims.userId,
      role: claims.role ?? 'user',
    };

    if (claims.name) payload['name'] = claims.name;
    if (claims.grants) payload['grants'] = claims.grants;

    return jwt.sign(payload, this.apiSecret, {
      issuer: this.apiKey,
      expiresIn: claims.expiresIn ?? '1h',
      algorithm: 'HS256',
    });
  }

  verify(token: string): jwt.JwtPayload {
    return jwt.verify(token, this.apiSecret, {
      issuer: this.apiKey,
      algorithms: ['HS256'],
    }) as jwt.JwtPayload;
  }
}
